app/src/main/java/co/condorlabs/customcomponents/customcollapsibleview/CollapsibleView.kt

```diff
@@ -0,0 +1,585 @@
+        textHiddenState =
+            typedArray.getString(R.styleable.CollapsibleView_collapsibleHiddenFooterText)
+                ?: resources.getString(R.string.hide)
+        textShowState = typedArray.getString(R.styleable.CollapsibleView_collapsibleShowFooterText)
+            ?: resources.getString(R.string.show)
+        setTitle(typedArray.getString(R.styleable.CollapsibleView_collapsibleTitle))
+        setSubtitle(typedArray.getString(R.styleable.CollapsibleView_collapsibleSubtitle))
+        setFooterTextColor(
+            typedArray.getColor(
+                R.styleable.CollapsibleView_collapsibleFooterTextColor,
+                NO_ID
+            )
+        )
+        setImageTint(typedArray.getColor(R.styleable.CollapsibleView_imageTintColor, NO_ID))
+        seeArrowIndicator(
+            typedArray.getBoolean(
+                R.styleable.CollapsibleView_seeArrowIndicator,
+                false
+            )
+        )
+
+        typedArray.recycle()
+    }
+
+    private fun collapseElements(requirementsHeight: Int) {
+        if (elementsAreCollapsed) {
+            collapseElements(ZERO, requirementsHeight, false)
+        } else {
+            collapseElements(requirementsHeight, ZERO, true)
+        }
+
+        elementsAreCollapsed = !elementsAreCollapsed
+
+        callOnCollapse(elementsAreCollapsed)
+    }
+
+    private fun collapseElements(from: Int, to: Int, isCollapsed: Boolean) {
+        updateFooterText(isCollapsed)
+
+        val animator =
+            ValueAnimator.ofInt(from, to).setDuration(DEFAULT_COLLAPSIBLE_ANIMATION_TIME)?.apply {
+                addUpdateListener {
+                    updateContainerViewHeight(it.animatedValue as Int)
+                    rotateRowIndicator(
+                        rotationValue(
+                            it.animatedValue as Int,
+                            if (from > to) from else to
+                        )
+                    )
+                }
+            } ?: return
+
+        startValueAnimator(animator)
+    }
+
+    private fun startValueAnimator(valueAnimator: ValueAnimator) {
+        AnimatorSet().apply {
+            play(valueAnimator)
+            interpolator = AccelerateDecelerateInterpolator()
+            start()
+        }
+    }
+
+    private fun updateContainerViewHeight(height: Int = ZERO) {
+        containerFrameLayout.apply {
+            layoutParams?.height = height
+            requestLayout()
+        }
+    }
+
+    private fun rotateRowIndicator(rotation: Float = ZERO_FLOAT) {
+        footerIndicatorImageView.rotation = -rotation
+    }
+
+    private fun rotationValue(currentValue: Int, limit: Int): Float {
+        return (COLLAPSIBLE_INDICATOR_ROTATION * ((currentValue * ONE_HUNDRED_FLOAT) / limit)) / ONE_HUNDRED_FLOAT
+    }
+
+    private fun updateFooterText(isCollapsed: Boolean) {
+        footerTextTextView.text = if (isCollapsed) textShowState else textHiddenState
```

Comment:
rename `textShowState` to `showStateText` and `textHiddenState` to hiddenStateText

Reply:
add {}

Reply:
Ok